src/sentry/api/event_search.py

```diff
@@ -387,6 +387,56 @@ def is_wildcard(self) -> bool:
             return False
         return bool(WILDCARD_CHARS.search(self.raw_value))
 
+    def classify_wildcard(self) -> Literal["prefix", "infix", "suffix", "other"]:
+        if not self.is_wildcard():
+            return "other"
+
+        ret = WILDCARD_CHARS.finditer(self.raw_value)
+
+        leading_wildcard = False
+        trailing_wildcard = False
+        middle_wildcard = False
+
+        for x in ret:
+            start, end = x.span()
+            if start == 0 and end == 1:
+                # It must span exactly [0, 1) because if it spans further,
+                # the pattern also matched on some leading slashes.
+                leading_wildcard = True
+            elif end == len(self.raw_value):
+                # It only needs to match on end because if it matches on
+                # some slashes before the *, that's okay.
+                trailing_wildcard = True
+            else:
+                middle_wildcard = True
+
+        if not middle_wildcard:
```

Comment:
To confirm I understand this, middle wildcard is something like
`foo*bar` which is a classification of wildcard we don't care about because we can't optimize it right?

Reply:
It's a little more generic than that. `middle_wildcard` is there to capture all cases where there's a `*` in the middle of the query like your example `foo*bar` or `f*o*o`. There's no easy way to optimize these types of wildcards, and they're less common so I choose to just not optimize them.